### ISSUE TITLE
Weird referenced-but-not-assigned errors due to arithmetic dereferencing

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1351]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
 [[ /sys/class/gpio/gpio4/value -eq "1" ]]
```

#### Here's what shellcheck currently says:
```
 [[ /sys/class/gpio/gpio4/value -eq "1" ]]
    ^-- SC2154: /sys/class/gpio/gpio4/value is referenced but not assigned.
```
#### Here's what I wanted or expected to see:

Something less confusing. This is not a valid variable name anyways.
